parquet-column/src/main/java/org/apache/parquet/column/statistics/SizeStatistics.java

```diff
@@ -0,0 +1,254 @@
+      definitionLevelHistogram = new long[maxDefinitionLevel + 1];
+      Arrays.fill(repetitionLevelHistogram, 0L);
+      Arrays.fill(definitionLevelHistogram, 0L);
+    }
+
+    /**
+     * Add repetition and definition level of a value to the statistics.
+     * It is called when value is null, or the column is not of BYTE_ARRAY type.
+     *
+     * @param repetitionLevel repetition level of the value
+     * @param definitionLevel definition level of the value
+     */
+    public void add(int repetitionLevel, int definitionLevel) {
+      Preconditions.checkArgument(0 <= repetitionLevel && repetitionLevel < repetitionLevelHistogram.length,
+        "repetitionLevel %s is out of range [0, %s]", repetitionLevel, repetitionLevelHistogram.length - 1);
+      Preconditions.checkArgument(definitionLevel < definitionLevelHistogram.length,
+        "definitionLevel %s is out of range [0, %s]", definitionLevel, definitionLevelHistogram.length - 1);
+      repetitionLevelHistogram[repetitionLevel]++;
+      definitionLevelHistogram[definitionLevel]++;
+    }
+
+    /**
+     * Add repetition and definition level of a value to the statistics.
+     * It is called when value is null, or the column is not of BYTE_ARRAY type.
+     *
+     * @param repetitionLevel repetition level of the value
+     * @param definitionLevel definition level of the value
+     * @param value value of to be added
+     */
+    public void add(int repetitionLevel, int definitionLevel, Binary value) {
+      add(repetitionLevel, definitionLevel);
+      if (type.getPrimitiveTypeName() == PrimitiveType.PrimitiveTypeName.BINARY) {
+        unencodedByteArrayDataBytes = Math.addExact(unencodedByteArrayDataBytes, value.length());
+      }
+    }
+
+    /**
+     * Build a SizeStatistics from the builder.
+     */
+    public SizeStatistics build() {
+      return new SizeStatistics(type, unencodedByteArrayDataBytes,
+        new LongArrayList(repetitionLevelHistogram), new LongArrayList(definitionLevelHistogram));
+    }
+
+  }
+
+  /**
+   * Create a builder to create a SizeStatistics.
+   *
+   * @param type physical type of the column associated with this statistics
+   * @param maxRepetitionLevel maximum repetition level of the column
+   * @param maxDefinitionLevel maximum definition level of the column
+   */
+  public static Builder newBuilder(PrimitiveType type, int maxRepetitionLevel, int maxDefinitionLevel) {
+    return new Builder(type, maxRepetitionLevel, maxDefinitionLevel);
+  }
+
+  /**
+   * Create a SizeStatistics.
+   *
+   * @param type physical type of the column associated with this statistics
+   * @param unencodedByteArrayDataBytes number of physical bytes stored for BYTE_ARRAY data values assuming no encoding
+   * @param repetitionLevelHistogram histogram for all repetition levels if non-empty
+   * @param definitionLevelHistogram histogram for all definition levels if non-empty
+   */
+  public SizeStatistics(PrimitiveType type,
```

Comment:
should this be private since we have the class builder?

Reply:
This is used to restore it from a serialized thrift object.